crates/holochain/src/conductor/conductor.rs

```diff
@@ -1169,13 +1208,980 @@ impl Conductor {
+                let network = cell.holochain_p2p_dna().clone();
+                match tokio::time::timeout(JOIN_NETWORK_TIMEOUT, network.join(cell_id.agent_pubkey().clone(), maybe_initial_arc)).await {
+                    Ok(Err(e)) => {
+                        tracing::info!(error = ?e, cell_id = ?cell_id, "Error while trying to join the network");
+                        Err(cell_id)
+                    }
+                    Err(_) => {
+                        tracing::info!(cell_id = ?cell_id, "Timed out trying to join the network");
+                        Err(cell_id)
+                    }
+                    Ok(Ok(_)) => Ok(cell_id),
+                }
+            });
+
+        let maybes: Vec<_> = futures::stream::iter(tasks)
+            .buffer_unordered(100)
+            .collect()
+            .await;
+
+        let (cell_ids, failed_joins): (Vec<_>, Vec<_>) =
+            maybes.into_iter().partition(Result::is_ok);
+
+        // These unwraps are both safe because of the partition.
+        let cell_ids: Vec<_> = cell_ids.into_iter().map(Result::unwrap).collect();
+        let failed_joins: Vec<_> = failed_joins.into_iter().map(Result::unwrap_err).collect();
+
+        // Update the status of the cells which were able to join the network
+        // (may or may not be all cells which were added)
+        self.update_cell_status(cell_ids.as_slice(), CellStatus::Joined);
+
+        self.update_cell_status(failed_joins.as_slice(), CellStatus::PendingJoin);
+
+        cell_ids
+    }
+
+    pub(crate) fn p2p_agents_db(&self, hash: &DnaHash) -> DbWrite<DbKindP2pAgents> {
+        self.spaces
+            .p2p_agents_db(hash)
+            .expect("failed to open p2p_agent_store database")
+    }
+
+    pub(crate) fn p2p_batch_sender(&self, hash: &DnaHash) -> tokio::sync::mpsc::Sender<P2pBatch> {
+        self.spaces
+            .p2p_batch_sender(hash)
+            .expect("failed to get p2p_batch_sender")
+    }
+
+    pub(crate) fn p2p_metrics_db(&self, hash: &DnaHash) -> DbWrite<DbKindP2pMetrics> {
+        self.spaces
+            .p2p_metrics_db(hash)
+            .expect("failed to open p2p_metrics_store database")
+    }
+
+    //       █████  ███                               █████             █████
+    //      ░░███  ░░░                               ░░███             ░░███
+    //    ███████  ████   █████  ████████   ██████   ███████    ██████  ░███████
+    //   ███░░███ ░░███  ███░░  ░░███░░███ ░░░░░███ ░░░███░    ███░░███ ░███░░███
+    //  ░███ ░███  ░███ ░░█████  ░███ ░███  ███████   ░███    ░███ ░░░  ░███ ░███
+    //  ░███ ░███  ░███  ░░░░███ ░███ ░███ ███░░███   ░███ ███░███  ███ ░███ ░███
+    //  ░░████████ █████ ██████  ░███████ ░░████████  ░░█████ ░░██████  ████ █████
+    //   ░░░░░░░░ ░░░░░ ░░░░░░   ░███░░░   ░░░░░░░░    ░░░░░   ░░░░░░  ░░░░ ░░░░░
+    //                           ░███
+    //                           █████
+    //                          ░░░░░
```

Comment:
I don't want to be the not-fun-guy... but I don't think I can get behind this. It's too hard to change, not searchable, and if we're feeling the need for such large callouts, let's just make some modules... this file could use some more breaking up anyways.